### ISSUE TITLE
This Session's transaction has been rolled back due to a previous exception during flush.

### ISSUE THREAD

**Issue by sanicki** [hjalp/leddit#2]
The script appears to run successfully, but then does not exit:

> Update complete. Sleeping for 3600 seconds.

However it eventually fails with:

> This Session's transaction has been rolled back due to a previous exception during flush.

If run again, the script recreates all posts and comments rather than only adding new posts and updating comments on the posts it previously created.

**Comment by hjalp** [comment]
I did a few local test runs over several hours using the subreddit you want to sync, and I wasn't able to replicate the transaction rollback exception. Is there a process that is shutting down the bot by force while it's making commits to the database?

> If run again, the script recreates all posts and comments rather than only adding new posts and updating comments on the posts it previously created.

This is caused by the default settings in `config.yaml`. The settings are optimized for active subreddits that receive around a hundred new posts a day, so the database is set to remove and stop updating posts that are older than the `max_post_age` to prevent it from growing too large. The default is 24 hours. The bot checks the database to see if a post has been posted before, and because your subreddit receives less than one new post a day, it removes all of the "old" posts from the database with each cycle and copies them over again.

There's no system to automatically calculate the update interval and total duration based on the original subreddit's activity yet, so it has to be set manually. This small subreddit would give better results with a `max_post_age` of `5184000` (60 days) and a `scrape_interval` of `86400` (once a day).

**Comment by sanicki** [comment]
> Is there a process that is shutting down the bot by force while it's making commits to the database?

No. My `scrape_interval` was set to the default `3600` and the error appears to correspond to that 60 minute mark.

> better results with a `max_post_age` of `5184000` (60 days) and a `scrape_interval` of `86400` (once a day).

I'll try this. I didn't realize that the script might better be run as a background process, like:

`nohup python3.11 main.py > log.txt &`

**Comment by sanicki** [comment]
Still happening.

The green indicates the original set of mirrored posts, which seem to be receiving new comments.

The red indicates the duplicate set of mirrored posts, which seem to appear each `scrape_interval`. 

![Screenshot 2023-07-12 10 01 37 AM (1)](https://github.com/hjalp/leddit/assets/6764844/574bc025-619e-40f4-a8d9-7aaf2d29d672)
